plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/MavenHostedFacet.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.maven;
+
+import javax.annotation.Nullable;
+
+import org.sonatype.nexus.repository.Facet;
+
+/**
+ * Maven hosted facet, present on all Maven hosted-type repositories.
+ *
+ * @since 3.0
+ */
+@Facet.Exposed
+public interface MavenHostedFacet
+    extends Facet
+{
+  /**
+   * Rebuilds/updates Maven metadata.
+   *
+   * @param update      if {@code true}, updates existing metadata, otherwise overwrites them with newly generated
+   *                    ones.
+   * @param groupId     scope the work to given groupId.
+   * @param artifactId  scope the work to given artifactId (groupId must be given).
+   * @param baseVersion scope the work to given baseVersion (groupId and artifactId must ge given).
+   */
+  void rebuildMetadata(boolean update,
```

Comment:
```OTOH, consider following use case: hosted repo, you deploy plugins org.foo:bar-maven-plugin and org.foo:baz-maven-plugin. And then you delete org.foo:bar-maven-plugin. In this case update=false would get rid of the deleted plugin only, update=true would still keep refs to it...```
Sounds like update=false should be the only thing we support in this case. Is there any compelling reason to have a code pathway that would allow us to be in a situation where we generate inconsistent data? Depending on callers to know the difference should only be something we support if there's a substantial benefit IMO